### PR TITLE
Upgrade to lucene-6.5.0-snapshot-f919485.

### PR DESCRIPTION
**Pull request by jpountz**
This is an in-progress PR as I have a failure in MatchQueryIT that looks related to recent changes around graph token streams. If someone could help me dig it, I would appreciate.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java [hunk 1]
```diff
@@ -172,6 +172,7 @@ public void testPhrasePrefix() throws ExecutionException, InterruptedException {
         assertSearchHits(searchResponse, "1", "2", "3", "7", "8");
     }
```
Comment (jimczi) [inline core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java]: Regarding the failing tests, this is due to a change in Lucene 6.5.0. In 6.4 Queries with multi word synonyms create a GraphQuery with all possible paths. So if you have "foo, foo bar" and a query like "foo boo" it will produce two boolean queries packed in a GraphQuery:
`foo boo` OR `foo bar boo`
Since we have all paths we can apply the cutoff frequency to each but in 6.5 the query parser will now create a query:
`(+foo OR (+foo +bar)) AND boo`
So the multi synonyms is now treated as a single term. This is better in terms of performance but it breaks some tests because the logic has changed. We cannot apply cutoff frequency on multi terms synonyms anymore (they are treated as a single term). Same problem with `minimum_should_match`.
IMO we should just rewrite this test and remove the multi terms synonym.

Comment (jpountz) [inline core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java]: ++

Comment (jpountz) [inline core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java]: However I'll leave it for a follow-up PR if you don't mind.

File: core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/*
```
Comment (jimczi) [inline core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java]: I think this deserves a separate PR, WDYT ? We need to document how this works and make sure that we add all the warnings regarding the restrictions of using this filter in conjunction with others.

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java]: I can do that. I added it because the test that checks whether all analysis components are exposed failed otherwise.

File: core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java (hunk 1 above)
Comment (jimczi) [inline core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java]: Sure, I can take over when you merged this PR. Thanks

File: core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java (hunk 2 above)
Comment (jimczi) [inline core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java]: Ok I created https://github.com/elastic/elasticsearch/issues/23104 to track the inclusion of this new filter. We can add the documentation and tests in a follow-up

**Review by jimczi — APPROVED** [review]
LGTM
